nats-base-client/types.ts

```diff
@@ -2658,31 +2680,68 @@ export type ConsumeOptions = ConsumeBytes | ConsumeMessages;
 export type PullConsumerOptions = FetchOptions | ConsumeOptions;
 
 export type MaxMessages = {
+  /**
+   * Maximum number of messages to retrieve.
+   * @default 100 messages
+   */
   max_messages: number;
 };
 
 export type MaxBytes = {
+  /**
+   * Maximum number of bytes to retrieve - note request must fit the entire message
+   * to be honored (this includes, subject, headers, etc). Partial messages are not
+   * supported.
+   */
   max_bytes: number;
 };
 
 export type ThresholdMessages = {
+  /**
+   * Threshold message count on which the client will auto-trigger additional requests
+   * from the server. This is only applicable to `consume`.
+   * @default  75% of {@link max_messages}.
+   */
   threshold_messages?: number;
 };
 
 export type ThresholdBytes = {
+  /**
+   * Threshold bytes on which the client wil auto-trigger additional message requests
+   * from the server. This is only applicable to `consume`.
+   * @default 75% of {@link max_bytes}.
+   */
   threshold_bytes?: number;
 };
 
 export type Expires = {
+  /**
+   * Amount of milliseconds to wait for messages before issuing another request.
+   * Note this value shouldn't be set by the user, as the default provides proper behavior.
+   *
+   * Minimum value is 1000 (1s).
+   * @default 30_000 (30s)
+   */
   expires?: number;
 };
 
 export type IdleHeartbeat = {
+  /**
+   * Number of milliseconds to wait for a server heartbeat when not actively receiving
+   * messages. If two or more heartbeats are missed in a row, the consumer will emit
+   * a notification. Note this value shouldn't be set by the user, as the default provides
+   * the proper behavior.
```

Comment:
Maybe instead, say it "does not have to be set b the user", and explain shortly what are the implications of low and high values?